runtime/ftplugin/idris2.vim

```diff
@@ -0,0 +1,27 @@
+" Vim ftplugin file
+" Language:	Idris2
+" Last Change:	2024 Nov 05
+" Maintainer:	Serhii Khoma <srghma@gmail.com>
+" License:             Vim (see :h license)
+" Repository:          https://github.com/ShinKage/idris2-nvim
+"
+" Based on ftplugin/idris2.vim from https://github.com/edwinb/idris2-vim
+
+if exists("b:did_ftplugin")
+  finish
+endif
+
+setlocal shiftwidth=2
+setlocal tabstop=2
+if exists('g:idris2#allow_tabchar') && g:idris2#allow_tabchar != 0
+  setlocal noexpandtab
+else
+  setlocal expandtab
+endif
+
+setlocal comments=s1:{-,mb:-,ex:-},:\|\|\|,:--
+setlocal commentstring=--\ %s
+setlocal iskeyword+=?
```

Comment:
why adding the `?`

Reply:
new feature of Idris2

https://raw.githubusercontent.com/idris-lang/Idris2/0659bcc2c0c8b437affd9d33b62668f0b08c74f7/docs/source/implementation/overview.rst

```
Additional type inference
-------------------------

A ``?`` in a type means "infer this part of the type".  This is distinct from
``_`` in types, which means "I don't care what this is". The distinction is in
what happens when inference fails.  If inference fails for ``_``, we implicitly
bind a new name (just like pattern matching on the lhs - i.e. it means match
anything). If inference fails for ``?``, we leave it as a hole and try to fill
it in later. As a result, we can say:

.. code-block:: idris

    foo : Vect ? Int
    foo = [1,2,3,4]

... and the ``?`` will be inferred to be 4. But if we say:

.. code-block:: idris

    foo : Vect _ Int
    foo = [1,2,3,4]

... we'll get an error, because the ``_`` has been bound as a new name.
Both ``?`` and ``_`` are represented in ``RawImp`` by the ``Implicit``
constructor, which has a boolean flag meaning "bind if unresolved".

So the meaning of ``_`` is now consistent on the lhs and in types (i.e. it
means infer a value and bind a variable on failure to infer anything). In
practice, using ``_`` will get you the old Idris behaviour, but ``?`` might
get you a bit more type inference.
```

Reply:
That doesn't sound like you need to add `?` as part of a word character.

Reply:
ah, right, maybe author meant it for named holes

```
vzipWith f [] [] = ?vzipWith_rhs_3
```

commented

Reply:
I see thanks